src/libcore/str/pattern.rs

```diff
@@ -304,7 +304,8 @@ unsafe impl<'a> Searcher<'a> for CharSearcher<'a> {
     fn next_match(&mut self) -> Option<(usize, usize)> {
         loop {
             // get the haystack after the last character found
-            let bytes = if let Some(slice) = self.haystack.as_bytes().get(self.finger..) {
+            let bytes = if let Some(slice) =self.haystack.as_bytes()
```

Comment:
nit: missing space after the `=` in `if let Some(slice) =self.haystack.as_bytes()`

Reply:
fixed